src/fs/visit.rs

```diff
@@ -56,15 +67,15 @@ pub fn discovery<Arc: ArcLookup>(arc: &Arc, path: &PathBuf, accepted: &mut HashM
 /// Files starting with a period are filtered out, and only the files with relevant regions are kept.  
 /// This signifies that if your goal is to simply get all the files, this is not the method to use.  
 /// This method exists to support backward compatibility with Ultimate Mod Manager.  
-pub fn umm_discovery<Arc: ArcLookup>(arc: &Arc, dir: &PathBuf, accepted: &mut HashMap<Hash40, ModFile>, rejected: &mut Vec<(PathBuf, RejectionReason)>) {
+pub fn umm_discovery<Arc: ArcLookup>(arc: &Arc, dir: &PathBuf, accepted: &mut HashMap<Hash40, ModFile>, rejected: &mut Vec<(PathBuf, RejectionReason)>, stream: &mut HashMap<Hash40, ModFile>) {
```

Comment:
I think a bonus of `DiscoveryResult` is that it'd unify and clean up these two function signatures a good bit.

If both umm_discovery and discovery need to be run, you could provide a merge/join function on `DiscoveryResult` and then do:

```rust
let files = discovery(...).merge(umm_discovery(...));
```